pkg/apis/configuration/validation/validation.go

```diff
@@ -184,15 +188,49 @@ func validateStatusMatch(s string, fieldPath *field.Path) field.ErrorList {
 	return allErrs
 }
 
-func validateStatusCodeInRange(status string) string {
-	code, err := strconv.ParseInt(status, 10, 64)
+func validateStatusCodeRange(statusRange []string) string {
+	min, msg := validateIntFromString(statusRange[0])
+	if msg != "" {
+		return msg
+	}
+
+	max, msg := validateIntFromString(statusRange[1])
+	if msg != "" {
+		return msg
+	}
+
+	for _, code := range statusRange {
+		if msg := validateStatusCodeInRange(code); msg != "" {
+			return msg
+		}
+	}
+
+	if max <= min {
+		return fmt.Sprintf("range limits must be %v < %v", min, max)
+	}
+
+	return ""
+}
+
+func validateIntFromString(number string) (int, string) {
+	numberInt, err := strconv.ParseInt(number, 10, 64)
 	if err != nil {
-		return fmt.Sprintf("%v must be a valid integer", status)
+		return 0, fmt.Sprintf("%v must be a valid integer", number)
+	}
+
+	return int(numberInt), ""
+}
+
+func validateStatusCodeInRange(status string) string {
```

Comment:
with this addition, we now have two functions with very similar names `validateStatusCodeInRange` and `validateStatusCodeRange`. For clarity, I suggest renaming `validateStatusCodeInRange` to `validateStatusCode`. So that `validateStatusCode` validates an individual status code and `validateStatusCodeRange` validates a range.

Reply:
That won't work, the check has to be done both in ranges and separate status codes. Renaming to `validateStatusCode` but checking that the code is in the range there too. It's still a validation of the status code so it will work.